### PR TITLE
don't fail login if exception raised from signal

### PR DESCRIPTION
**Pull request by snopoke**
@TylerSheffels

### CODE REVIEW

File: corehq/util/decorators.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from functools import wraps
+import logging
+from corehq.util.global_request import get_request
+from dimagi.utils.logging import notify_exception
+
+
+def handle_uncaught_exceptions(mail_admins=True):
```
Comment (snopoke) [inline corehq/util/decorators.py]: @dimagi/team-commcare-hq FYI

File: corehq/util/decorators.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    """Decorator to log uncaught exceptions and prevent them from
+    bubbling up the call chain.
+    """
+    def _outer(fn):
+        @wraps(fn)
+        def _handle_exceptions(*args, **kwargs):
+            try:
+                return fn(*args, **kwargs)
+            except Exception as e:
```
Comment (dimagimon) [inline corehq/util/decorators.py]: F841 local variable 'e' is assigned to but never used

File: corehq/util/decorators.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                msg = "Uncaught exception from {}.{}".format(fn.__module__, fn.__name__)
```
Comment (TylerSheffels) [inline corehq/util/decorators.py]: @snopoke Do you also want to include the exception message in the error?

Comment (esoergel) [inline corehq/util/decorators.py]: I believe `notify_exception` includes the most recent exception automagically.

Comment (snopoke) [inline corehq/util/decorators.py]: That's correct, the exception info is added automatically when using
logging.exception

**Pull request merged**